AWSSDRegistry crash on nil Endpoint Labels map

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

-->

**What happened**:
Running `docker.io/bitnami/external-dns:0.7.5-debian-10-r0` AWS EKS, only `--source=node` with `--provider=aws-sd --registry=aws-sd` crashed on initial loop:
```{"level":"info","msg":"Instantiating new Kubernetes client","time":"2021-05-20T12:30:38Z"}
{"level":"info","msg":"Using inCluster-config based on serviceaccount-token","time":"2021-05-20T12:30:38Z"}
{"level":"info","msg":"Created Kubernetes client https://172.20.0.1:443","time":"2021-05-20T12:30:38Z"}
panic: assignment to entry in nil map

goroutine 1 [running]:
sigs.k8s.io/external-dns/registry.(*AWSSDRegistry).updateLabels(0xc000353100, 0xc0000fc0a0, 0x1, 0x1)
/bitnami/blacksmith-sandox/external-dns-0.7.5/src/github.com/kubernetes-incubator/external-dns/registry/aws_sd_registry.go:86 +0x94
sigs.k8s.io/external-dns/registry.(*AWSSDRegistry).ApplyChanges(0xc000353100, 0x2b13320, 0xc0000debd0, 0xc0002ff200, 0xc0000fc088, 0x1)
/bitnami/blacksmith-sandox/external-dns-0.7.5/src/github.com/kubernetes-incubator/external-dns/registry/aws_sd_registry.go:76 +0x245
sigs.k8s.io/external-dns/controller.(*Controller).RunOnce(0xc0006bd320, 0x2b13320, 0xc0000debd0, 0x3ca5ae0, 0x2440101)
/bitnami/blacksmith-sandox/external-dns-0.7.5/src/github.com/kubernetes-incubator/external-dns/controller/controller.go:154 +0x3d6
sigs.k8s.io/external-dns/controller.(*Controller).Run(0xc0006bd320, 0x2b13260, 0xc0001082c0)
/bitnami/blacksmith-sandox/external-dns-0.7.5/src/github.com/kubernetes-incubator/external-dns/controller/controller.go:191 +0x20f
main.main()
/bitnami/blacksmith-sandox/external-dns-0.7.5/src/github.com/kubernetes-incubator/external-dns/main.go:353 +0xe30
```

**What you expected to happen**:
external-dns will create new service with instances there without issues.
```
{"level":"info","msg":"Instantiating new Kubernetes client","time":"2021-05-20T12:41:34Z"}
{"level":"info","msg":"Using inCluster-config based on serviceaccount-token","time":"2021-05-20T12:41:34Z"}
{"level":"info","msg":"Created Kubernetes client https://172.20.0.1:443","time":"2021-05-20T12:41:34Z"}
{"level":"info","msg":"Creating a new service \"service\" in \"ns-xx\" namespace","time":"2021-05-20T12:41:42Z"}
{"level":"info","msg":"Registering a new instance \"54…\" for service \"service\" (srv-pzf)","time":"2021-05-20T12:41:42Z"}
{"level":"info","msg":"Registering a new instance \"3…\" for service \"service\" (srv-pzf)","time":"2021-05-20T12:41:42Z"}
{"level":"info","msg":"All records are already up to date","time":"2021-05-20T12:44:42Z"}
```

**How to reproduce it (as minimally and precisely as possible)**:
Using Cloud Map public DNS Namespace, force external-dns to register new Cloud Map Service for a `--source=node`, new Endpoint would be created with "default" Labels map, as a result registry will try to [set owner](https://github.com/kubernetes-sigs/external-dns/blob/master/registry/aws_sd_registry.go#L76) but will fail with `panic: assignment to entry in nil map`.

**Anything else we need to know?**:
Running external-dns v0.7.5 against already created Cloud Map Service works fine, as it will already have a description field.
Also tested with `v0.8.0-8`, – same result.

**Environment**:
- External-DNS version (use `external-dns --version`): `v0.7.5`
- DNS provider: `aws-sd`
- Others: `--source=node`

--- a/registry/aws_sd_registry.go
+++ b/registry/aws_sd_registry.go
@@ -83,6 +83,9 @@ func (sdr *AWSSDRegistry) ApplyChanges(ctx context.Context, changes *plan.Change
 
 func (sdr *AWSSDRegistry) updateLabels(endpoints []*endpoint.Endpoint) {
 	for _, ep := range endpoints {
+		if ep.Labels == nil {
+			ep.Labels = make(map[string]string)
+		}
 		ep.Labels[endpoint.OwnerLabelKey] = sdr.ownerID
 		ep.Labels[endpoint.AWSSDDescriptionLabel] = ep.Labels.Serialize(false)
 	}

--- a/source/node.go
+++ b/source/node.go
@@ -151,6 +151,7 @@ func (ns *nodeSource) Endpoints(ctx context.Context) ([]*endpoint.Endpoint, erro
 		}
 
 		ep.Targets = endpoint.Targets(addrs)
+		ep.Labels = endpoint.NewLabels()
 
 		log.Debugf("adding endpoint %s", ep)
 		if _, ok := endpoints[ep.DNSName]; ok {

--- a/source/node_test.go
+++ b/source/node_test.go
@@ -307,6 +307,19 @@ func testNodeSourceEndpoints(t *testing.T) {
 			},
 			false,
 		},
+		{
+			"node with nil Lables returns valid endpoint",
+			"",
+			"",
+			"node1",
+			[]v1.NodeAddress{{Type: v1.NodeExternalIP, Address: "1.2.3.4"}},
+			nil,
+			map[string]string{},
+			[]*endpoint.Endpoint{
+				{RecordType: "A", DNSName: "node1", Targets: endpoint.Targets{"1.2.3.4"}, Labels: map[string]string{}},
+			},
+			false,
+		},
 	} {
 		t.Run(tc.title, func(t *testing.T) {
 			// Create a Kubernetes testing client

--- a/source/shared_test.go
+++ b/source/shared_test.go
@@ -20,6 +20,7 @@ import (
 	"sort"
 	"strings"
 	"testing"
+	"reflect"
 
 	"sigs.k8s.io/external-dns/endpoint"
 )
@@ -59,5 +60,10 @@ func validateEndpoint(t *testing.T, endpoint, expected *endpoint.Endpoint) {
 	// if non-empty record type is expected, check that it matches.
 	if expected.RecordType != "" && endpoint.RecordType != expected.RecordType {
 		t.Errorf("expected %s, got %s", expected.RecordType, endpoint.RecordType)
+	}
+
+	// if non-empty labels are expected, check that they matches.
+	if expected.Labels != nil && !reflect.DeepEqual(endpoint.Labels,expected.Labels) {
+		t.Errorf("expected %s, got %s", expected.Labels, endpoint.Labels)
 	}
 }

